compiler/rustc_target/src/spec/windows_gnu_base.rs

```diff
@@ -86,6 +87,10 @@ pub fn opts() -> TargetOptions {
         emit_debug_gdb_scripts: false,
         requires_uwtable: true,
         eh_frame_header: false,
+        // FIXME(davidtwco): Support Split DWARF on Windows GNU - may require LLVM changes to
+        // output DWO, despite using DWARF, doesn't use ELF..
+        debuginfo_kind: DebuginfoKind::Pdb,
+        supported_split_debuginfo: Cow::Borrowed(&[SplitDebuginfo::Off]),
```

Comment:
@wesleywiser is this correct? We spoke about Split DWARF technically being possible on Windows GNU before, but I find that if I say `debuginfo_kind: DebuginfoKind::Dwarf` then LLVM complains that the target can't output a DWO because it doesn't use ELF.

Reply:
I'm not super familiar with `*-windows-gnu` but from what I can see, it looks like rustc generates a single output file (ie, there is no `.dwo` or `.pdb`) and that output file contains embedded DWARF debuginfo. 

Eg:

```
$ rustc --target x86_64-pc-windows-gnu -g --crate-name hello_world_gnu hello_world.rs
$ ls | grep hello_world_gnu
hello_world_gnu.exe
$ objdump -W hello_world_gnu.exe 
hello_world_gnu.exe:     file format pei-x86-64

...
Contents of the .debug_info section:

...
  Compilation Unit @ offset 0x2948:
   Length:        0xad (32-bit)
   Version:       4
   Abbrev Offset: 0x5a2
   Pointer Size:  8
 <0><2953>: Abbrev Number: 1 (DW_TAG_compile_unit)
    <2954>   DW_AT_producer    : (indirect string, offset: 0x21f): clang LLVM (rustc version 1.65.0-nightly (0b79f758c 2022-08-18))
    <2958>   DW_AT_language    : 28     (Rust)
    <295a>   DW_AT_name        : (indirect string, offset: 0x260): hello_world.rs\@\hello_world_gnu.2deb01f1-cgu.0
    <295e>   DW_AT_stmt_list   : 0x486
    <2962>   DW_AT_comp_dir    : (indirect string, offset: 0x290): D:\code\temp
    <2966>   DW_AT_GNU_pubnames: 1
    <2966>   DW_AT_low_pc      : 0x140001530
    <296e>   DW_AT_high_pc     : 0x5
...
```

Reply:
Right, when I set `DebuginfoKind::Dwarf` for that target, then we got this error..

```
LLVM ERROR: dwo only supported with ELF and Wasm
```

..which makes sense when your output shows..

```
hello_world_gnu.exe:     file format pei-x86-64
```

..so it looks like Split DWARF would _technically_ be possible on Windows GNU, but LLVM would need to support DWARF objects on PEI first, and then I would need to extend `thorin` to package DWARF in PEI. But what all of that does mean, is that despite windows-gnu using DWARF, I can't set it to `DebuginfoKind::Dwarf` (there's probably still something up because `DebuginfoKind::Dwarf` and `supported_split_debuginfo: &[SplitDebuginfo::Off]` shouldn't make LLVM do anything).

Reply:
Reading Appendix F in the DWARF 5 spec, I don't see why an ELF binary would be required. It seems like the appropriate sections could be placed in the PE file the same way the DWARF is but I could be missing something.

Reply:
> Reading Appendix F in the DWARF 5 spec, I don't see why an ELF binary would be required. It seems like the appropriate sections could be placed in the PE file the same way the DWARF is but I could be missing something.

Yeah, I don't think it is required by DWARF, just seems like a LLVM restriction.

Reply:
On an unrelated note, I think we could probably support `SplitDebuginfo::Unpacked` on `*-windows-mscv` via the [`/DEBUG:FASTLINK` option](https://docs.microsoft.com/en-us/cpp/build/reference/debug-generate-debug-info?view=msvc-170#:~:text=The%20/DEBUG%3AFASTLINK%20option,build%20products%20available.) which sounds pretty much like the equivalent split DWARF behavior. I'll put that on my todo list to look into.